reporting/registry.go

```diff
@@ -0,0 +1,36 @@
+package reporting
+
+import (
+	"log"
+	"reflect"
+
+	metrics "github.com/rcrowley/go-metrics"
+)
+
+// RegisterMetric tag support for metrics.Register()
+func RegisterMetric(name string, metric interface{}, tags map[string]string) {
+	key := EncodeKey(name, tags)
+	metrics.Register(key, metric)
```

Comment:
`metrics.Register` returns an error as well. Let's check for error and return that too.